src/goto-programs/goto_program.cpp

```diff
@@ -668,6 +668,65 @@ bool goto_programt::instructiont::equals(const instructiont &other) const
   // clang-format on
 }
 
+bool goto_programt::instructiont::check_internal_invariants(
+  const symbol_tablet &table,
+  messaget &msg) const
+{
+  bool found_violation = false;
+  std::vector<std::string> id_collector;
+  auto symbol_finder = [&](const exprt &e) {
+    if(e.id() == ID_symbol)
+    {
+      auto symbol_expr = to_symbol_expr(e);
+      if(!table.has_symbol(symbol_expr.get_identifier()))
+        id_collector.push_back(id2string(symbol_expr.get_identifier()));
```

Comment:
We could probably also use the `DATA_CHECK()` macros here as in #3128. I was thinking if we want a third method where we keep going and report all errors we could add a third validation mode, say `validation_modet::KEEP_GOING` which does not throw an exception or yield a `DATA_INVARIANT` violation but instead prints the error messages to the `error()` stream. But that's future work I'd say.